src/test/java/org/example/UserTest.java

```diff
@@ -26,133 +20,109 @@
     }
 
     @Test
-    void postUserWithIncorrectMailFormat() throws IOException {
-        CloseableHttpClient httpclient = HttpClients.createDefault();
-        HttpPost httpPost = new HttpPost(String.format("https://gorest.co.in/public/v2/users?access-token=%s", token));
-        List<NameValuePair> params = new ArrayList<NameValuePair>(2);
-        params.add(new BasicNameValuePair("name", "Mary"));
-        params.add(new BasicNameValuePair("mail", "marygmail.com"));
-        params.add(new BasicNameValuePair("gender", "female"));
-        params.add(new BasicNameValuePair("status", "active"));
-        httpPost.setEntity(new UrlEncodedFormEntity(params));
-        Header header = new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
-        httpPost.setHeader(header);
-        HttpResponse response = httpclient.execute(httpPost);
-        Assert.assertEquals(response.getCode(), 422);
+    void testCreateUserWithIncorrectMailFormat() throws IOException, ParseException {
+        User user = testDataFactory.generateTestUser();
+        user.setEmail("johnmail.com");
+        CloseableHttpResponse response = httpService.sendPostRequest(user);
+        int responseCode = response.getCode();
+        Assert.assertEquals(responseCode, HTTP_UNPROCESSABLECONTENT, "Wrong response code. Expected: " + HTTP_UNPROCESSABLECONTENT + " Received: " + responseCode);
     }
 
     @Test
-    public void patchUser() throws IOException {
-        CloseableHttpClient client = HttpClients.createDefault();
-        HttpPatch httpPatch = new HttpPatch(String.format("https://gorest.co.in/public/v2/users/7887693?access-token=%s", token));
-        List<NameValuePair> params = new ArrayList<NameValuePair>(2);
-        params.add(new BasicNameValuePair("name", "John Doe"));
-        params.add(new BasicNameValuePair("email", String.format("%s@gmail.com", RandomStringUtils.random(10, true, true))));
-        params.add(new BasicNameValuePair("gender", "male"));
-        params.add(new BasicNameValuePair("status", "active"));
-        httpPatch.setEntity(new UrlEncodedFormEntity(params));
-        Header header = new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + token);
-        httpPatch.setHeader(header);
-        String response = client.execute(httpPatch, new BasicHttpClientResponseHandler());
-        List<ValidationMessage> errors = validatePayload(response, "patchUserTemplate");
-        Assert.assertTrue(errors.isEmpty());
+    public void testPatchUser() throws IOException, ParseException {
+        CloseableHttpResponse response = httpService.sendPatchRequest(testDataFactory.generateTestUser());
+        int responseCode = response.getCode();
+        String responseBody = EntityUtils.toString(response.getEntity());
+        Assert.assertEquals(responseCode, HTTP_OK, "Wrong response code. Expected: " + HTTP_OK + " Received: " + responseCode);
```

Comment:
You only check status code and JSON schema validation, but do not verify if the patch took effect, the returned user reflects the changes you sent.